### PR TITLE
Add service-worker for in-between session ETag management

### PR DESCRIPTION
**Pull request by TimvdLippe**
This PR adds a service-worker for in-between session ETag management. Caching on the client during a session remains if your browser does not support service-worker. The biggest advantage is when you refresh a lot of times where the cache of `authenticated-ajax` is cleaned, but the `service-worker` cache is persistent.

Following the description in the README to fix the certificate in your browser.

### CODE REVIEW

File: certificate/server.key [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (GijsWeterings) [inline certificate/server.key:1]: BAD! even for self-signed certs ;)

File: src/home-page.html [hunk 2]
```diff
@@ -39,7 +39,7 @@
 
     <h1 class="name">PReview Code</h1>
     <div class="logo-container">
-      <img class="logo" src="http://blueseptember.org.au/wp-content/uploads/2015/07/Blue-square.jpg" />
+      <img class="logo" src="https://blueseptember.org.au/wp-content/uploads/2015/07/Blue-square.jpg" />
```
Comment (GijsWeterings) [inline src/home-page.html:42]: best logo ever, by the way

File: certificate/server.key (hunk 1 above)
Comment (TimvdLippe) [inline certificate/server.key:1]: For `localhost` certificates I thought this was safe. My reference was `browsersync`: https://github.com/BrowserSync/browser-sync/blob/master/lib/server/certs/server.crt

File: src/home-page.html (hunk 2 above)
Comment (TimvdLippe) [inline src/home-page.html:42]: :blue_heart:

File: certificate/server.key (hunk 1 above)
Comment (GijsWeterings) [inline certificate/server.key:1]: In itself it is, but it could expose future weaknesses if the cert/passphrase is mishandled

**Comment by txsmith** [comment]
It seems to work! Also, kudos for pushing a private RSA key

**Comment by TimvdLippe** [comment]
Pull request updated to work properly with 304 status of the GitHub API (by actually returning the previous response).

**Comment by eanker** [comment]
Whilst running bower update:

`Please note that,
    rite-evaluation depends on carbon-route#reset-tail which resolved to carbon-route#ee79d74c68
    carbon-elements#0.1.0 depends on carbon-route#^0.8.0 which resolved to carbon-route#0.8.5
Resort to using carbon-route#reset-tail which resolved to carbon-route#ee79d74c68
Code incompatibilities may occur.


Please note that,
    rite-evaluation depends on marked-element#expose-callback which resolved to marked-element#25791b727b
    iron-demo-helpers#1.2.3, iron-doc-viewer#1.0.14 depends on marked-element#^1.0.0 which resolved to marked-element#1.1.3
Resort to using marked-element#expose-callback which resolved to marked-element#25791b727b
Code incompatibilities may occur.


Please note that,
    rite-evaluation depends on prism-element#fork which resolved to prism-element#fcb9dda2ef
    iron-demo-helpers#1.2.3 depends on prism-element#^1.0.0 which resolved to prism-element#1.0.3
    iron-doc-viewer#1.0.14 depends on prism-element#^1.0.2 which resolved to prism-element#1.0.3
Resort to using prism-element#fork which resolved to prism-element#fcb9dda2ef
Code incompatibilities may occur.
`

**Pull request merged**